velero-plugins/common/util.go

```diff
@@ -29,3 +32,56 @@ func HasImageRefPrefix(s, prefix string) bool {
 	}
 	return refSplit[0] == prefix
 }
+
+// LocalImageReference describes an image in the internal openshift registry
+type LocalImageReference struct {
+	Registry  string
+	Namespace string
+	Name      string
+	Tag       string
+	Digest    string
+}
+
+// ParseLocalImageReference
+func ParseLocalImageReference(s, prefix string) (*LocalImageReference, error) {
+	refSplit := strings.Split(s, "/")
+	if refSplit[0] != prefix {
+		return nil, fmt.Errorf("image reference is not local")
+	}
+	if len(refSplit) != 3 {
+		return nil, fmt.Errorf("Unexpected image reference %s", s)
+	}
+	parsed := LocalImageReference{Registry: prefix, Namespace: refSplit[1]}
+	digestSplit := strings.Split(refSplit[2], "@")
+	if len(digestSplit) > 2 {
+		return nil, fmt.Errorf("Unexpected image reference %s", s)
+	} else if len(digestSplit) == 2 {
+		parsed.Name = digestSplit[0]
+		parsed.Digest = digestSplit[1]
+		return &parsed, nil
+	}
+	tagSplit := strings.Split(refSplit[2], ":")
+	if len(tagSplit) > 2 {
+		return nil, fmt.Errorf("Unexpected image reference %s", s)
+	} else if len(tagSplit) == 2 {
+		parsed.Tag = tagSplit[1]
+	}
+	parsed.Name = tagSplit[0]
+	return &parsed, nil
+}
+
+// SwapContainerImageRefs updates internal image references from
+// backup registry to restore registry pathnames
+func SwapContainerImageRefs (containers []corev1API.Container, oldRegistry, newRegistry string, log logrus.FieldLogger) {
+	for n, container := range containers {
+		imageRef := container.Image
+		log.Infof("container image ref %s", imageRef)
+		newImageRef, err := ReplaceImageRefPrefix(imageRef, oldRegistry, newRegistry)
+		if err == nil {
+			// Replace local image
+			log.Infof("replacing container image ref %s with %s", imageRef, newImageRef)
```

Comment:
NIT: Curious if we should make this `log.Infof("[util] replacing...")`. Only because the log statements prior to this call specify where they are coming from.

Honestly I'm indifferent just something I noticed